### ISSUE TITLE
Extend possible enabled addon states, not just true/false

### ISSUE THREAD

**Issue by WorldLanguages** [ScratchAddons/ScratchAddons#5431]
### Why this would be helpful/solve a problem

As of now, addons can only be enabled or disabled. When the user installs for the first time, or when an update adds new addons that didn't previously exist, we check at the value of `enabledByDefault` and that's all. After the extension is restarted, we have no idea if the user disabled an addon, or if the user never tried it out. There's little information we store on `chrome.storage`.

### What this change does

Instead of false/true, we could use the following possibilities:
1. `default`: the user didn't explicitly opt-in or opt-out into the feature. The value of `enabledByDefault` on the manifest would be respected. We might want to add a new possible property on addon manifests named `enabledOnInstall` to recover the old behavior of `enabledByDefault` in case we need it (for example, in case we want to enable an addon by default for new users, but not for existing ones).
2. `disabled`: the addon is explicitly disabled.
3. `enabled`: the addon is explicitly enabled.
4. `recommendedEnabled`: the user enabled a feature (which doesn't exist yet) such as "automatically enable all recommended addons", and a recommended addon wasn't previously enabled manually. This would allow us to show these addons in a different way in the settings page, so the user can know the addon has been automatically enabled, and not manually enabled.

The concept of `recommendedEnabled` could be expanded for other concepts, such as "enabled because of a superpreset", or it could be extended for other addon tags.

Additionally, no matter the enabled state stored in `chrome.storage`, if we add the ability to disable addons remotely for all users (#5248), the addon should be logically disabled. There are also other reasons we might want to allow addons to be logically enabled/disabled during the span of the session (in all tabs or a specific one) such as addon tours (#5407)

It may also be useful to store the date where the user last changed the enabled value of an addon. We could also store a log of historical changes (instead of just the most recent one per each addon), but I'll leave that for another issue. Knowing whether a user recently tested out an addon can be valuable information at the time of considering what addons to suggest to the user.

### Possible alternatives

There's a lot of possible discussion here on different ways we could do this.
Any alternative is probably better than the current true/false model.

### Demonstration(s) and additional context

This concept could also be extended to addon settings, potentially.

**Comment by DNin01** [comment]
I think it would be cool to see this help us with "personalized" addon suggestions and recently used lists. But also, IMO, users should be aware of the fact that the addons they're using are being tracked, even if Scratch Addons doesn't send any data to databases or servers for analytics or other things.

**Comment by BroJac5246** [comment]
> I think it would be cool to see this help us with "personalized" addon suggestions and recently used lists. But also, IMO, users should be aware of the fact that the addons they're using are being tracked, even if Scratch Addons doesn't send any data to databases or servers for analytics or other things.

Calling that "tracking" is debatable. We might as well mention it in the Privacy Policy, though.

**Comment by DogCatPuppyLover** [comment]
> I think it would be cool to see this help us with "personalized" addon suggestions and recently used lists. But also, IMO, users should be aware of the fact that the addons they're using are being tracked, even if Scratch Addons doesn't send any data to databases or servers for analytics or other things.

Since Scratch Addons stores your extension preferences anyway, and it's not sending this information to other servers (except maybe browser companies for syncing purposes?), I don't think it's a privacy problem. Tracking is when another server is collecting data on how you're using a service.

**Comment by WorldLanguages** [comment]
It wouldn't be a serious a privacy problem, but we have a section in the privacy policy where we mention some things that are stored locally, so we could mention any potential new stored information there.

**Comment by penguinmoose** [comment]
> I think it would be cool to see this help us with "personalized" addon suggestions and recently used lists. But also, IMO, users should be aware of the fact that the addons they're using are being tracked, even if Scratch Addons doesn't send any data to databases or servers for analytics or other things.

Like DogCatPuppyLover said, I wouldn't consider that tracking, because we actually don't know what addons people enable. Even though it is stored, that information isn't getting sent anywhere - it simply stays on your device. However, we can mention it in the privacy policy.

**Comment by Secret-chest** [comment]
> However, we can mention it in the privacy policy.

No need to even do that - we already store settings.json lol

**Comment by DNin01** [comment]
> > However, we can mention it in the privacy policy.
> 
> No need to even do that - we already store settings.json lol

That's a little different. Users will know that any app will save and remember settings and customizations, so that doesn't really need a mention, plus, Scratch Addons doesn't currently use it for any purpose other than keeping addons enabled and configured, all of which is visible and obvious to the user. But users may not know that it is recording user activity, if we implement that. That's why I touched on mentioning _that_ in the privacy policy. It may not be absolutely necessary to point out that users will be watched if the plan is to not collect this data, though.

**Comment by WorldLanguages** [comment]
Yeah, enough talk about privacy concerns. If anyone has something more to say about that, please open a different issue/discussion.

**Comment by iqnite** [comment]
A possible alternative would be to only have true/false but also have the addons "remember" if they were last toggled by the user or by default.

**Comment by WorldLanguages** [comment]
> A possible alternative would be to only have true/false but also have the addons "remember" if they were last toggled by the user or by default.

Yeah, we would still need to use booleans *somewhere* in memory, as an addon needs to either run or not (and quantum computers aren't a thing yet) - but in storage, we could go with `"enabled"` or `"disabled"` if we want.

**Comment by DNin01** [comment]
> go with `"enabled"` or `"disabled"` if we want

May I suggest using `"on"` and `"off"` to save space in storage?

**Comment by WorldLanguages** [comment]
Right, because of the storage limit we probably want to use enums if possible.
